### PR TITLE
Add lein-scss as an option for CSS preprocessing

### LINKED ISSUE #135: Add lein-scss option
Hi there,

Currently learning my way around re-frame and thought adding a [lein-scss](https://github.com/bluegray/lein-scss) option alongside lein-less would be a good idea.
If it is, I would be happy to take a stab at it.

### PR DESCRIPTION
**Pull request by pcasaretto**
Fixes #135 

Opening it as a draft to get early feedback.
What I did was basically search for all occurrences of less and adapted them to scss.
I'm still getting `Error: invalid profile(s)` when I try to use it.
Haven't been able to figure out why. I will try harder as soon as I can.

### CODE REVIEW

File: src/leiningen/new/options/scss.clj [hunk 1]
```diff
@@ -0,0 +1,7 @@
+(ns leiningen.new.options.scss
+  (:require [leiningen.new.options.helpers :as helpers]))
+
+(def option "scss")
```
Comment (superstructor) [inline src/leiningen/new/options/scss.clj:4]: This needs to be `(def option "+scss")` (i.e. missing `+`). Fixes `Error: invalid profile(s)`.

File: src/leiningen/new/re_frame/project.clj [hunk 2]
```diff
@@ -35,21 +36,33 @@
...
+                                   :pretty-print? true}}]}
+ {{/garden?}}{{#less?}}
   :less {:source-paths ["less"]
          :target-path  "resources/public/css"}
-{{/less?}}
-
-  :shell {:commands {"open" {:windows ["cmd" "/c" "start"]
-                             :macosx  "open"
-                             :linux   "xdg-open"}}}
+ {{/less?}{{#scss?}}
```
Comment (superstructor) [inline src/leiningen/new/re_frame/project.clj:49]: `/less?` closing tag is missing a closing `}`. i.e. should be ` {{/less?}}{{#scss?}}`. Causes `clojure.lang.ExceptionInfo: Unclosed tag: less? at line 49, column 10`.

File: src/leiningen/new/re_frame/project.clj [hunk 3]
```diff
@@ -35,21 +36,33 @@
...
+  :scss {:builds
+         {:develop    {:source-dir "scss/"
+                       :dest-dir   "resources/public/css/"
+                       :executable "sassc"
+                       :args       ["-m" "-I" "scss/" "-t" "nested"]}
+          :production {:source-dir "scss/"
+                       :dest-dir   "resources/public/css/"
+                       :dest-dir   "public/css/"
```
Comment (superstructor) [inline src/leiningen/new/re_frame/project.clj:57]: One of these `:dest-dir` lines (line 56 or 57) needs to go as it causes a duplicate key syntax error when running the lein build (e.g. `lein dev`).

File: src/leiningen/new/re_frame/project.clj [hunk 4]
```diff
@@ -35,21 +36,33 @@
...
+                       :executable "sassc"
+                       :args       ["-I" "scss/" "-t" "compressed"]
+                       :jar        true}}}}
```
Comment (superstructor) [inline src/leiningen/new/re_frame/project.clj:60]: One too many closing `}`. Causes unmatched delimiter `}` syntax error when running the lein build (e.g. `lein dev`).

**Comment by pcasaretto** [comment]
Thanks for the help @superstructor !
Managed to get it working.
Need your input on two things:
1. lein-scss defines profiles. For now, I defined two profiles (`:dev` and `:prod`). Dev outputs sourcemaps and uncompressed output, while prod outputs code suitable for production. All `prep` tasks associated with scss are currently using `:dev`, and I'm not sure if that's corrent.
2. lein-scss depends on an external executable. The template currently mentions sassc. Should document this and alert the user on install or find a way to automatically install that?

**Comment by pcasaretto** [comment]
Hey @superstructor , just a friendly ping here.

**Comment by superstructor** [comment]
@pcasaretto 
Re 1. if the prep task is run for both dev and prod aliases then using dev profile for both is probably wrong. Alternative might be to not use prep tasks and just inline the call to lein-scss in each alias. The repetition that results is OK. 

Re 2. just document it imo in general the tooling ecosystem we're working with here doesn't provide for auto-installing system-level dependencies; e.g. `npm` etc.

**Comment by pcasaretto** [comment]
Closing as I haven't had the time to continue investing in it.

**Pull request closed without merging**